nequip/data/AtomicData.py

```diff
@@ -249,6 +250,44 @@ def from_ase(cls, atoms, r_max, **kwargs):
             **add_fields,
         )
 
+    def to_ase(self) -> Union[List[ase.Atoms], ase.Atoms]:
+        """Build a (list of) ``ase.Atoms`` object(s) from an ``AtomicData`` object.
+
+        For each unique batch number associated with AtomicDataDict.BATCH_KEY,
+        an ``ase.Atoms`` object is created. If AtomicDataDict.BATCH_KEY does not
+        exist in self, a single ``ase.Atoms`` object is created.
+
+        Returns:
+            A list of ``ase.Atoms`` objects if AtomicDataDict.BATCH_KEY is in self
+            and is not None. Otherwise, a single ``ase.Atoms`` object is returned.
+        """
+        positions = self.pos
+        atomic_nums = self.atomic_numbers
+        pbc = self.pbc if AtomicDataDict.PBC_KEY in self else None
+        cell = self.cell if AtomicDataDict.CELL_KEY in self else None
+        batch = self.batch if AtomicDataDict.BATCH_KEY in self else None
+
+        if batch is not None:
+            unique_batches = range(batch.max() + 1)
+            batch_atoms = []
+            for batch_idx in unique_batches:
+                mask = batch == batch_idx
+                mol = Atoms(
+                    numbers=atomic_nums[mask],
+                    positions=positions[mask],
+                    cell=cell[batch_idx] if cell is not None else None,
+                    pbc=pbc[batch_idx] if pbc is not None else None,
+                )
+                batch_atoms.append(mol)
+            return batch_atoms
+        else:
+            return Atoms(
+                numbers=atomic_nums,
+                positions=positions,
+                cell=cell[0] if cell is not None else None,
+                pbc=pbc[0] if pbc is not None else None,
```

Comment:
this special case applies here too

you can handle it by doing something like:
```
cell = cell.view(-1, 3, 3)
pbc = pbc.view(-1, 3)
```
which forces the -1 dimension to exist, even if of size 1

this would be before L270

Reply:
Would I have to check if `cell` and `pbc` are `None` beforehand? (If so, would it be bad that I check that again when I create the `ase.Atoms` objects)?

Reply:
yes, you would, and no harm in checking it twice. unfortunately python doesnt provide much more elegant ways to do it